apps/src/sharedFunctionalBlocks.js

```diff
@@ -348,7 +468,7 @@ function installCond(blockly, generator, numPairs) {
   generator[blockName] = function() {
     var cond, value, defaultValue;
     var code = 'function() {\n  ';
-    for (var i = 0; i < numPairs; i++) {
+    for (var i = 0; i <= this.numPairs_; i++) {
```

Comment:
Actually, you're right. this.numPairs_ isnt a thing. Instead it should be this.pairs_.length, and we should be using i to index into that. Looks like a test case I should try to cover too.